odoo/tests/common.py

```diff
@@ -248,10 +248,13 @@ def _assertRaises(self, exception):
             with self.env.clear_upon_failure():
                 yield cm
 
-    def assertRaises(self, exception, func=None, *args, **kwargs):
-        if func:
+    def assertRaises(self, exception, *args, **kwargs):
+        if not hasattr(self, 'env'):
+            return super().assertRaises(exception, *args, **kwargs)
+        if args:
+            # args[0] is a callable
             with self._assertRaises(exception):
-                func(*args, **kwargs)
+                args[0](*args[1:], **kwargs)
```

Comment:
/!\ you change the method's signature in a stable version and make the assumption that all callsites pass func as a positional arg and not as a kwarg /!\